upup/pkg/fi/cloudup/awstasks/autoscalinggroup.go

```diff
@@ -55,11 +55,11 @@ type AutoscalingGroup struct {
 	// LaunchTemplate is the launch template for the asg
 	LaunchTemplate *LaunchTemplate
 	// MaxSize is the max number of nodes in asg
-	MaxSize *int64
```

Comment:
Thanks for doing this! I think we should keep these types as integers (or int64) and only update the cloudformation (line 865) to string. We shouldn't modify the terraform types because those should use integers (see the terraform docs [here](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group)).

Only modifying the `cloudformationAutoscalingGroup` type also means we don't have to update anything in `pkg/model` which I think is preferable for this particular issue.

This same suggestion applies to the loadbalancer and healthcheck types below

Reply:
@rifelpet , the fields in the loadbalancer and health check also results in lint warnings,
should they be reverted back to int64 fields ?

```
E3012 Property Resources/AWSElasticLoadBalancingLoadBalancerapiprivateciliumadvancedexamplecom/Properties/Listeners/0/InstancePort should be of type String
E3012 Property Resources/AWSElasticLoadBalancingLoadBalancerapiprivateciliumadvancedexamplecom/Properties/Listeners/0/LoadBalancerPort should be of type String
E3012 Property Resources/AWSElasticLoadBalancingLoadBalancerapiprivateciliumadvancedexamplecom/Properties/HealthCheck/HealthyThreshold should be of type String
E3012 Property Resources/AWSElasticLoadBalancingLoadBalancerapiprivateciliumadvancedexamplecom/Properties/HealthCheck/UnhealthyThreshold should be of type String
E3012 Property Resources/AWSElasticLoadBalancingLoadBalancerapiprivateciliumadvancedexamplecom/Properties/HealthCheck/Interval should be of type String
E3012 Property Resources/AWSElasticLoadBalancingLoadBalancerapiprivateciliumadvancedexamplecom/Properties/HealthCheck/Timeout should be of type String
```

```
E3012 Property Resources/AWSElasticLoadBalancingLoadBalancerbastionprivateciliumadvancedexamplecom/Properties/Listeners/0/InstancePort should be of type String
tests/integration/update_cluster/privateciliumadvanced/cloudformation.json:1297:13

E3012 Property Resources/AWSElasticLoadBalancingLoadBalancerbastionprivateciliumadvancedexamplecom/Properties/Listeners/0/LoadBalancerPort should be of type String

```

Reply:
These types should be updated to fix the lint warnings:

https://github.com/kubernetes/kops/blob/0be8d4e3258f10204327dfb0eddbc2dc20affe3d/upup/pkg/fi/cloudup/awstasks/load_balancer.go#L789-L829

but the regular `LoadBalancer`, `LoadBalancerHealthCheck` types should not. This means the type conversion from int64 to string will need to happen in `RenderCloudformation`. The `./hack/update-expected.sh` script should only be updating the `cloudformation.json` files, not the `kubernetes.tf` files.